pkg/bootstrap/platform/gcp_test.go

```diff
@@ -183,13 +209,77 @@ func TestGCPMetadata(t *testing.T) {
 			for e, v := range tt.env {
 				os.Setenv(e, v)
 			}
-			once = sync.Once{}
-			mg := gcpEnv{tt.shouldFill, tt.projectIDFn, tt.numericProjectIDFn, tt.locationFn, tt.clusterNameFn, tt.instanceIDFn,
-				tt.instanceTemplateFn, tt.instanceCreatedByFn}
-			got := mg.Metadata()
+			shouldFillMetadata, projectIDFn, numericProjectIDFn, clusterLocationFn, clusterNameFn, instanceNameFn, instanceIDFn, instanceTemplateFn, createdByFn =
+				tt.shouldFill, tt.projectIDFn, tt.numericProjectIDFn, tt.locationFn, tt.clusterNameFn, tt.instanceNameFn, tt.instanceIDFn,
+				tt.instanceTemplateFn, tt.instanceCreatedByFn
+			e := NewGCP()
+			got := e.Metadata()
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("gcpEnv.Metadata() => '%v'; want '%v'", got, tt.want)
+			}
+			os.Clearenv()
```

Comment:
nit: are we sure we want to clear all environment variables?  can we just reset the one we care about?